src/wh_server_nvm.c

```diff
@@ -0,0 +1,311 @@
+        resp.rc = 0;
+
+        /* Convert the response struct */
+        wh_MessageNvm_TranslateSimpleResponse(magic,
+                &resp, (whMessageNvm_SimpleResponse*)resp_packet);
+        *out_resp_size = sizeof(resp);
+    }; break;
+
+    case WH_MESSAGE_NVM_ACTION_LIST:
+    {
+        whMessageNvm_ListRequest req = {0};
+        whMessageNvm_ListResponse resp = {0};
+
+        /* Convert request struct */
+        wh_MessageNvm_TranslateListRequest(magic,
+                (whMessageNvm_ListRequest*)req_packet, &req);
+
+        /* Process the list action */
+        resp.rc = wh_Nvm_List(server->nvm,
+                req.access, req.flags, req.startId,
+                &resp.count, &resp.id);
+
+        /* Convert the response struct */
+        wh_MessageNvm_TranslateListResponse(magic,
+                &resp, (whMessageNvm_ListResponse*)resp_packet);
+        *out_resp_size = sizeof(resp);
+    }; break;
+
+    case WH_MESSAGE_NVM_ACTION_GETAVAILABLE:
+    {
+        /* No Request packet */
+        whMessageNvm_GetAvailableResponse resp = {0};
+
+        /* Process the available action */
+        resp.rc = wh_Nvm_GetAvailable(server->nvm,
+                &resp.avail_size, &resp.avail_objects,
+                &resp.reclaim_size, &resp.reclaim_objects);
+
+        /* Convert the response struct */
+        wh_MessageNvm_TranslateGetAvailableResponse(magic,
+                &resp, (whMessageNvm_GetAvailableResponse*)resp_packet);
+        *out_resp_size = sizeof(resp);
+    }; break;
+
+    case WH_MESSAGE_NVM_ACTION_GETMETADATA:
+    {
+        whMessageNvm_GetMetadataRequest req = {0};
+        whMessageNvm_GetMetadataResponse resp = {0};
+        whNvmMetadata meta = {0};
+
+        /* Convert request struct */
+        wh_MessageNvm_TranslateGetMetadataRequest(magic,
+                (whMessageNvm_GetMetadataRequest*)req_packet, &req);
+
+        /* Process the getmetadata action */
+        resp.rc = wh_Nvm_GetMetadata(server->nvm, req.id, &meta);
+
+        if (resp.rc == 0) {
+            resp.id = meta.id;
+            resp.access = meta.access;
+            resp.flags = meta.flags;
+            resp.len = meta.len;
+            memcpy(resp.label, meta.label, sizeof(resp.label));
+        }
+        /* Convert the response struct */
+        wh_MessageNvm_TranslateGetMetadataResponse(magic,
+                &resp, (whMessageNvm_GetMetadataResponse*)resp_packet);
+        *out_resp_size = sizeof(resp);
+    }; break;
+
+    case WH_MESSAGE_NVM_ACTION_ADDOBJECT:
+    {
+        whMessageNvm_AddObjectRequest req = {0};
+        uint16_t hdr_len = sizeof(req);
+        whNvmMetadata meta = {0};
+        const uint8_t* data = (const uint8_t*)req_packet + hdr_len;
+        whMessageNvm_SimpleResponse resp = {0};
+
+        /* Convert request struct */
+        wh_MessageNvm_TranslateAddObjectRequest(magic,
+                (whMessageNvm_AddObjectRequest*)req_packet, &req);
+
+        if(req_size == (hdr_len + req.len)) {
```

Comment:
Should we be checking `req_size` for every case? It looks like this is the only place we even use it. If we never are going to use it then we should probably remove it as an argument.

Reply:
Yes.  This is another miss.  req_size should always be checked to ensure the received packet is of the right size.  I'll add those checks.